module/x/peggy/abci.go

```diff
@@ -20,50 +20,49 @@ func slashing(ctx sdk.Context, k keeper.Keeper) {
 	params := k.GetParams(ctx)
 	currentBondedSet := k.StakingKeeper.GetBondedValidatorsByPower(ctx)
 
-	// valsets are sorted so the most recent one is first
-	valsets := k.GetValsets(ctx)
-	if len(valsets) == 0 {
-		k.SetValsetRequest(ctx)
-	}
+	// valsets are sorted by nonce in ASC order
+	unslashedValsets := k.GetUnSlashedValsets(ctx)
 
-	for i, vs := range valsets {
+	for _, vs := range unslashedValsets {
 		signedWithinWindow := uint64(ctx.BlockHeight()) > params.SignedValsetsWindow && uint64(ctx.BlockHeight())-params.SignedValsetsWindow > vs.Height
```

Comment:
I think this variable name is a little confusing, change it or add a comment. Maybe 'withinSlashingWindow?' becuase it's not actually checking if it's signed it's checking if it's within the slashing window.